### PR TITLE
Cherry-pick to 7.10: docs: update generate_fields_docs.py (#21359)

### PR DESCRIPTION
**Pull request by bmorelli25**
Backports the following commits to 7.10:
 - docs: update generate_fields_docs.py (#21359)

### CODE REVIEW

**Review by dedemorton — APPROVED** [review]
LGTM

**Comment by bmorelli25** [comment]
@elastic/uptime, I don't have merge privileges until one of you approves this; can someone take a look when they have a sec?

**Review by justinkambic — APPROVED** [review]
LGTM, because this seems a routine change that is being applied across all beats and my team was the only one required for CODEOWNER review.

**Pull request merged**
